### PR TITLE
added the api for trip details, created the tripdetailpane with all information

### PR DESCRIPTION
**Pull request by aaronbrethorst**
- [x] Left align back button and add button text on trip details (see screenshot)
- [x] Use real data for the departure times shown on the right side of the pane (see screenshot)
- [x] Fill in the current location of the bus, if available. See iOS equivalent code here: https://github.com/OneBusAway/onebusaway-ios/blob/main/OBAKit/Trip/TripStopListItem.swift#L73
- [x] Fill in the user icon, too. iOS code here: https://github.com/OneBusAway/onebusaway-ios/blob/main/OBAKit/Trip/TripStopListItem.swift#L66

image

### CODE REVIEW

File: src/components/oba/TripDetailsPane.svelte [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	async function loadTripDetails() {
+		try {
+			let url = `/api/oba/trip-details/${tripId}?includeTrip=true&includeSchedule=true&includeStatus=true`;
+			if (serviceDate) {
+				url += `&serviceDate=${serviceDate}`;
+			}
+			const response = await fetch(url);
+			if (response.ok) {
+				const data = await response.json();
+				tripDetails = data.data.entry;
```
Comment (aaronbrethorst) [inline src/components/oba/TripDetailsPane.svelte:49]: I suggest extracting `data.data` into its own local variable and using that in place of `data.data` throughout this code:

```
const jsonBody = await response.json();
const data = jsonBody.data;
```

File: src/components/oba/TripDetailsPane.svelte [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+				if (data.data.references && data.data.references.routes) {
```
Comment (aaronbrethorst) [inline src/components/oba/TripDetailsPane.svelte:51]: now that you have `data` extracted above, you can use that along with [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) to simplify these conditions:

```
if (data?.references?.routes) {
  // ...
}
```

File: src/components/oba/TripDetailsPane.svelte [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+					routeInfo = data.data.references.routes.find((route) => route.id === tripDetails.routeId);
+				}
+
+				if (data.data.references && data.data.references.stops) {
```
Comment (aaronbrethorst) [inline src/components/oba/TripDetailsPane.svelte:55]: here too

File: src/components/oba/TripDetailsPane.svelte [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+					stopInfo = data.data.references.stops.reduce((acc, stop) => {
+						acc[stop.id] = stop;
+						return acc;
+					}, {});
+				}
+
+				if (tripDetails.status && tripDetails.status.closestStop) {
```
Comment (aaronbrethorst) [inline src/components/oba/TripDetailsPane.svelte:62]: and this becomes `if (tripDetails.status?.closestStop)`

File: src/components/oba/TripDetailsPane.svelte [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+					currentStopIndex = tripDetails.schedule.stopTimes.findIndex(
+						(stop) => stop.stopId === tripDetails.status.closestStop.stopId
+					);
+				}
+
+				calculateBusPosition();
+
+				console.log('Trip details:', tripDetails);
```
Comment (aaronbrethorst) [inline src/components/oba/TripDetailsPane.svelte:70]: are these still necessary? if not, please delete them

**Review by aaronbrethorst — COMMENTED** [review]
looking really good! I have noted a number of things I'd like to see changed before merging, but I love how effective you've been in tackling this task!

File: src/components/oba/StopPane.svelte [hunk 6]
```diff
@@ -96,19 +120,70 @@
...
+							on:showTripDetails={handleShowTripDetails}
 						/>
 					{/each}
 				</div>
 			</div>
 		</div>
 	{/if}
+
+	{#if showTripDetails}
+		<div class="trip-details-modal scrollbar-hidden">
```
[the remark this answers is not in the document]
Comment (tarunsinghofficial) [reply src/components/oba/StopPane.svelte:135]: Sure. Done.

File: src/components/oba/TripDetailsPane.svelte [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+	}
+
+	async function loadTripDetails() {
+		try {
+			let url = `/api/oba/trip-details/${tripId}?includeTrip=true&includeSchedule=true&includeStatus=true`;
+			if (serviceDate) {
+				url += `&serviceDate=${serviceDate}`;
+			}
+			const response = await fetch(url);
+			if (response.ok) {
```
[the remark this answers is not in the document]
Comment (tarunsinghofficial) [reply src/components/oba/TripDetailsPane.svelte:47]: Done with the recommended changes.

File: src/components/oba/TripDetailsPane.svelte (hunk 1 above)
Comment (tarunsinghofficial) [reply src/components/oba/TripDetailsPane.svelte:49]: Extracted the data into the local variable.

File: src/components/oba/TripDetailsPane.svelte (hunk 5 above)
Comment (tarunsinghofficial) [reply src/components/oba/TripDetailsPane.svelte:70]: No! Deleted all logs.

**Review by aaronbrethorst — COMMENTED** [review]
nice work @tarunsinghofficial. I made a few more tweaks and am merging. thanks for your hard work on this!

**Comment by tarunsinghofficial** [comment]
> nice work @tarunsinghofficial. I made a few more tweaks and am merging. thanks for your hard work on this!

Nice UI enhancement on Tripdetailspane. Thanks for the changes.

**Pull request merged**